components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2096,6 +2096,26 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         });
     }
 
+    async getNotifications(ctx: TraceContext): Promise<string[]> {
+        const result = await super.getNotifications(ctx);
+        const user = this.checkAndBlockUser("getNotifications");
+        if (user.usageAttributionId) {
+            const allSessions = await this.listBilledUsage(ctx, user.usageAttributionId);
+            const totalUsage = allSessions.map((s) => s.credits).reduce((a, b) => a + b, 0);
```

Comment:
Ideally we'd get that from the service as well, but we can replace that once we do.